library/CMakeLists.txt

```diff
@@ -195,6 +195,7 @@ foreach(target IN LISTS target_libraries)
     target_include_directories(${target}
         PUBLIC ${MBEDTLS_DIR}/include/
         PUBLIC ${thirdparty_inc_public}
+        PRIVATE ${MBEDTLS_DIR}/library/
```

Comment:
As this is for the sake of 3rdparty code that we add this include directory I would rather add it to thirdparty_inc in 3rdparty/CMakeLists.txt.

Reply:
I don't understand why this shouldn't be in `library/CMakeLists.txt`. If we added other directories with library source files — say we had a `library/generated-from-fstar` and a `library/manually-written-constant-time` — they would all use the same include paths. Even now, the only reason why `library` doesn't need to be in the include path when compiling `library/*.c` is that the current directory is implicitly in the include path. If we moved `library/*.h` to a different directory as you suggest, we'd need to have it in a `target_include_directories` clause in `library/CMakeLists.txt`.

Reply:
> 
> 
> If we moved `library/*.h` to a different directory as you suggest, we'd need to have it in a `target_include_directories` clause in `library/CMakeLists.txt`.
Not necessarily, if the include headers are in ./library/internal and you include them with the internal/*.h pattern you don't need to add a new include directory, the current directory as a possible include directory (which it is) would be enough.

Reply:
Otherwise, what you are proposing is indeed correct. The only drawback I see is that the reason why the ./library directory is explicitly included (3rdparty code) is not explained and thus people could wonder why. If you prefer to keep it that way and not go the  ${thirdparty_inc} way then please add a comment. It would be nice to have such a comment in the make file as well.

Reply:
I still don't see a reason to move the inclusion of `library` out of `library/*ake*`. I've added comments.

Moving header files out of `library` is out of scope in this pull request.

Reply:
> 
> I still don't see a reason to move the inclusion of `library` out of `library/*ake*`. I've added comments.

As I already said, fine by me, thanks for adding comments and it looks good to me like that.

Reply:
> 
> Moving header files out of `library` is out of scope in this pull request.

yep